modules/lang-painless/src/main/java/org/elasticsearch/painless/symbol/SemanticScope.java

```diff
@@ -340,6 +354,12 @@ public Variable defineVariable(Location location, Class<?> type, String name, bo
     public abstract boolean isVariableDefined(String name);
     public abstract Variable getVariable(Location location, String name);
 
+    public void setUsesInstanceMethod() {}
```

Comment:
What about throwing an exception here because it's not valid to ever set it accidentally? What you've got is fine now but it'd be kind to the next person working on it if we said "no".

Reply:
There are lots of cases where we'll call this but the surrounding scope is not an LambdaScope, eg `byte get() {Byte.valueOf(5)} get()` in `FunctionTests.testReturnsAreUnboxedIfNeeded`.

In that case the scopes are `BlockScope` wrapped with a parent `FunctionScope`.

Reply:
Cool. Could you leave a comment saying that it's ok that this is a noop? Just so I don't get confused when I look again in a few months.

Reply:
Done.

Reply:
Thanks you kindly!